priv/http.js

```diff
@@ -0,0 +1,79 @@
+try { module.exports = {http:http}; } catch (e) { }
+
+// Template: http.send(url + '?' + 'test1=1&test2=2', 'GET', '', {SomeHeader:'some header'}).done(function(data, status, headers) {console.log(data););
+
+var http = {
+    receiveData:null,
+    doneCallback:function(){},
+    failCallback:function(){},
+    settings:{},
+    send: function(settings) {
+        var $ = this;
+        var defSett = {
+            url:window.location.href,
+            method:'GET',
+            body:'',
+            headers:{},
+            returnType:'html',
+        };
+        $.settings = {};
+        if (typeof settings === 'string') {
+            $.settings = $.merge(defSett, {url:settings});
+        } else {
+            $.settings = $.merge(defSett, settings);
+        }
+        if (/^\/\//.test($.settings.url)) {
+            $.settings.url = window.location.protocol + $.settings.url
+        }
+        if (!/^http[s]*\:\/\//.test($.settings.url)) {
+            $.settings.url = window.location.origin + $.settings.url
+        }
+        var tList = [];
+        if ($.settings.headers) {
+            for (var prop in $.settings.headers) {
+                tList.push(tuple(bin(prop),bin($.settings.headers[prop])));
+            }
+        }
+        ws.send(enc(tuple(atom('http'),
+            bin($.settings.url),
+            bin($.settings.method||'GET'),
+            bin($.settings.body||''),
+            tList)
+        ));
+        return $;
+    },
+    back: function(d, s, h) {
+        if (s >= 400) {
+            this.failCallback(d, s, h);
+        } else {
+            if (this.settings.returnType === 'json') {
+                try {
+                    d = JSON.parse(this.escapeSpecialChars(d));
```

Comment:
Actually yes, but it's just in case if json is not valid at all. I saw something similar somewhere... We can remove this escaping if it really  interferes and just to drop a error in console.